jcommune-model/src/main/resources/org/jtalks/jcommune/migrations/V73__Convert_Database_To_UTF8MB4_Encoding.sql

```diff
@@ -0,0 +1,39 @@
+DELIMITER $$
+DROP PROCEDURE IF EXISTS changeCollation$$
+-- Stored procedure to convert all tables character set and collations
+CREATE PROCEDURE changeCollation(IN character_set VARCHAR(255), IN collation_type VARCHAR(255))
+  BEGIN
+    DECLARE is_finished INTEGER DEFAULT 0;
+    DECLARE table_name varchar(255) DEFAULT "";
+    DECLARE t_cursor CURSOR FOR SELECT DISTINCT TABLE_NAME
+                                    FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_SCHEMA = DATABASE()
+                                                                          AND TABLE_TYPE = "BASE TABLE";
+    DECLARE CONTINUE HANDLER
+    FOR NOT FOUND SET is_finished = 1;
+    OPEN t_cursor;
+
+    get_table: LOOP
+      FETCH t_cursor INTO table_name;
+      IF is_finished = 1 THEN
+        LEAVE get_table;
+      END IF;
+-- Flyway cannot convert schema_version table by self because it uses 'SELECT FOR UPDATE' query inside the lib.
+      IF (table_name != '' AND table_name != 'jcommune_schema_version') THEN
+
+        SET @s = CONCAT('ALTER TABLE ', table_name,
+                        ' CONVERT TO CHARACTER SET ', character_set, ' COLLATE ', collation_type);
+        PREPARE stmt FROM @s;
+        EXECUTE stmt;
+        DEALLOCATE PREPARE stmt;
+        SET table_name = '';
+
+      END IF;
+    END LOOP get_table;
+    CLOSE t_cursor;
+  END $$
+DELIMITER ;
+
+ALTER DATABASE jtalks CHARACTER SET 'utf8mb4' COLLATE 'utf8mb4_unicode_ci';
```

Comment:
The database can be named differently on different envs.. This line will fail in most of cases.